### PR TITLE
Add CloudFormation Outputs for Datadog links

### PR DESCRIPTION
**Pull request by JacksonDavenport**
### What does this PR do?
Add CloudFormation Outputs for each Function in the CloudFormation Template, with a value of the URL linking to the corresponding function in Datadog.

### Motivation
Help navigation into the Datadog platform relative to these functions

### Testing Guidelines
Validated SAM and CDK deployments that this macro add the Outputs as appropriate, and will display them (amongst other outputs) with the native notation.

### Types of changes

- [ ] Bug fix
- [X] New feature
- [ ] Breaking change
- [ ] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [ ] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [ ] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog

### CODE REVIEW

**Review by tianchu — APPROVED** [review]

File: serverless/src/index.ts [hunk 1]
```diff
@@ -161,6 +175,25 @@ function lambdaHasDynamicallyGeneratedName(lambdas: LambdaFunction[]) {
...
+ * and value (URL) to the function in Datadog
+ */
+function addOutputLinks(outputs: Outputs, lambdas: LambdaFunction[], site: string) {
+  for (const lambda of lambdas) {
+    const functionKey = lambda.key;
+    const key = `DatadogMonitor${functionKey}`.replace(/[^a-z0-9]/gi, "");
+    // Create Fn::Sub string using the Logical ID of the function and AWS::Region/AccountId pseudoparameters
+    // https://app.datadoghq.com/functions/${LogicalID}:${AWS::Region}:${AWS::AccountId}:aws?source=cfn-macro
+    outputs[key] = {
+      Description: `See ${functionKey} in Datadog`,
```
Comment (Hesperide) [inline serverless/src/index.ts:189]:
```suggestion
      Description: `Monitor ${functionKey} in Datadog:`,
```

File: serverless/src/index.ts [hunk 2]
```diff
@@ -161,6 +175,25 @@ function lambdaHasDynamicallyGeneratedName(lambdas: LambdaFunction[]) {
...
 }
 
+/**
+ * Builds the CloudFormation Outputs containing the alphanumeric key, description,
+ * and value (URL) to the function in Datadog
+ */
+function addOutputLinks(outputs: Outputs, lambdas: LambdaFunction[], site: string) {
+  for (const lambda of lambdas) {
+    const functionKey = lambda.key;
+    const key = `DatadogMonitor${functionKey}`.replace(/[^a-z0-9]/gi, "");
```
Comment (Hesperide) [inline serverless/src/index.ts:185]: Are we able to include hyphens here? If so, could we do something like

`DatadogServerlessView-${functionKey}`

Comment (JacksonDavenport) [reply serverless/src/index.ts:185]: Unfortunately no, only alphanumeric characters are allowed. Can change it to `DatadogServerlessView${functionKey}` if that is preferred though.

File: serverless/src/index.ts (hunk 1 above)
Comment (JacksonDavenport) [reply serverless/src/index.ts]: 👍  updated

**Pull request merged**